src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -30,119 +30,85 @@
+ *     BLOCKS.register(FMLJavaModLoadingContext.get().getModEventBus());
+ * }
+ * </pre></code>
  *
  * @param <T> The base registry type
  */
+@SuppressWarnings("deprecation") // TODO 1.20.2+ - Find+Replace RegistryObject with DeferredHolder.
 public class DeferredRegister<T>
 {
     /**
-     * DeferredRegister factory for forge registries that exist <i>before</i> this DeferredRegister is created.
+     * Factory method for DeferredRegister. The target registry does not need to exist yet.<br>
      * <p>
-     * If you have a supplier, <u>do not use this method.</u>
-     * Instead, use one of the other factories that takes in a registry key or registry name.
-     *
-     * @param reg the forge registry to wrap
-     * @param modid the namespace for all objects registered to this DeferredRegister
-     * @see #create(ResourceKey, String)
-     * @see #create(ResourceLocation, String)
+     * If the registry is never created, any created {@link DeferredHolder}s will never become {@linkplain DeferredHolder#isPresent() present}.
+     * <p>
+     * This method requires a ResourceKey for a registry - how these are obtained varies by registry.<br>
+     * For {@linkplain IForgeRegistry forge registries}, use {@link IForgeRegistry#getRegistryKey()}.<br>
+     * For {@linkplain Registry vanilla registries}, use the static fields on {@link BuiltInRegistries}.<br>
+     * Additionally, the names of vanilla registries can be retrieved from {@link BuiltInRegistries#REGISTRY}.
+     * @param key The key of the registry to register objects to.
+     * @param modid The namespace for all objects registered to this DeferredRegister.
      */
-    public static <B> DeferredRegister<B> create(IForgeRegistry<B> reg, String modid)
+    public static <B> DeferredRegister<B> create(ResourceKey<? extends Registry<B>> key, String modid)
     {
-        return new DeferredRegister<>(reg, modid);
+        return new DeferredRegister<>(key, modid);
     }
 
     /**
-     * DeferredRegister factory for custom forge registries or {@link BuiltInRegistries vanilla registries}
-     * to lookup based on the provided registry key. Supports both registries that already exist or do not exist yet.
-     * <p>
-     * If the registry is never created, any {@link RegistryObject}s made from this DeferredRegister will throw an exception.
-     * To allow the optional existence of a registry without error, use {@link #createOptional(ResourceKey, String)}.
-     *
-     * @param key the key of the registry to reference. May come from another DeferredRegister through {@link #getRegistryKey()}.
-     * @param modid the namespace for all objects registered to this DeferredRegister
-     * @see #createOptional(ResourceKey, String)
-     * @see #create(IForgeRegistry, String)
-     * @see #create(ResourceLocation, String)
+     * @deprecated Use {@link #create(ResourceKey, String)} via {@link IForgeRegistry#getRegistryKey()}
```

Comment:
Missed a comment here: I have the same thoughts here as I did in my review, which is that we should point users to use `ForgeRegistries.Keys` instead, and deprecate most of the keys in that class.